### PR TITLE
Removes DESIGN.textile doc and update ARCHITECTURE.md

### PR DESCRIPTION
**Pull request by groyoh**
#### Purpose
Rebase #1490 and add missing changes.

#### Changes
Removes the `DESIGN.textile` file and adds references to the 0.8 and 0.9 READMES within the `ARCHITECTURE.md` file.

#### Additional helpful information

Closes #1490.

### CODE REVIEW

File: docs/ARCHITECTURE.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 [Back to Guides](README.md)
 
+This document focuses on architecture the 0.10.x version of ActiveModelSerializers. If you are interested in the architecture of the 0.8 or 0.9 versions,
+please refer to the [0.8 README](https://github.com/rails-api/active_model_serializers/blob/0-8-stable/README.md) or
+[0.9 README](https://github.com/rails-api/active_model_serializers/blob/0-9-stable/README.md).
```
Comment (bf4) [inline docs/ARCHITECTURE.md]: @groyoh Thanks.  We also want to reference the original design docs which was the impetus for this PR https://github.com/rails-api/active_model_serializers/pull/1490#issuecomment-178665552

Comment (groyoh) [inline docs/ARCHITECTURE.md]: Added it below ;) Let me know if something is missing.

File: docs/ARCHITECTURE.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
...
+
+The original design is also available [here](https://github.com/rails-api/active_model_serializers/blob/0-9-stable/DESIGN.textile).
```
Comment (bf4) [inline docs/ARCHITECTURE.md]: I'd rather link to one of the original docs 

- `[serializers.textile in Rails (before reverted)](https://github.com/rails/rails/blob/fcacc6986ab60f1fb2e423a73bf47c7abd7b191d/railties/guides/source/serializers.textile)` and
- `[README.textile in first 0.0.1 commit](https://github.com/rails-api/active_model_serializers/commit/d72b66d4c5355b0ff0a75a04895fcc4ea5b0c65e?short_path=8aa641a#diff-8aa641a619b8ebeca0116d41774daf45)`

Comment (groyoh) [inline docs/ARCHITECTURE.md]: Changed it to the first commit.

**Comment by groyoh** [comment]
I'll kill Travis, since it's only documentation change.

**Comment by bf4** [comment]
Or add [ci skip] to the commit

**Comment by groyoh** [comment]
Holy sheep! I didn't know this trick.

**Comment by groyoh** [comment]
@bf4 any other thoughts on this?

**Comment by remear** [comment]
This looks good to me.

**Pull request merged**